### DISCUSSION TITLE
[Postprocessor][Material property][getMaterialProperty]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#18545] [Q&A General] [answered]
Dear moose users,

How can I load material property in postprocessors?

Typically, I want to use NodalVariablePostprocessor to calculate some values by nodal variable and material property.

But, when I use getMaterialProperty<Real>, it doesn't work and it shows the following error:

 _'getMaterialProperty' was not declared in this scope; did you mean 'ADMaterialProperty'?_

I guess I have to add some **_header file that includes the definition of getMaterialProperty_**.

But I am not sure.

Kind regards

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes, in general, you have to add the MaterialPropertyInterface to the classes that the postprocessor inherits from in order to use the getMaterialProperty routine. This is done for the user object classes that a lot of postprocessors inherit from. 

However, you are using a Nodal postprocessor, and currently NodalUserObject does not inherit from that class. The reason is that this interface is not able to compute nodal values, as material properties are tied to subdomains, so it would not know why one to pick at an *interface between two subdomains*.

So the solution here is to wait for us to implement functor material properties, which is underway in #18395, and will do nodal values of material properties

OR 
 convert from nodal to elemental variables and use the Elemental postprocessors. 

Guillaume

**Reply by coskrrb2002** [reply]
Dear @GiudGiud, 

Thank you for your comment!

It helps me a lot.

I will look and follow up #18395 post.

Cheers,
Nakkyu
